### PR TITLE
Fixing redirect bug in index page

### PR DESCRIPTION
**Pull request by freddyucv**
### Proposed Changes
* Change the REGEX for Root Index Page

https://github.com/dotCMS/core/pull/25254/files#diff-fc6a10f4ecffde4f4501664220835f7b8b1413428561497ca3205fd711f146b8R27

* Improve the way we match the Domain and Port inside the REGEX

https://github.com/dotCMS/core/pull/25254/files#diff-0e580f38c58875e09d3247c7fd86090d1237652e2c5e257a53a8a27a799fbb6aR24-R26

* Improve the Index Page REGEX (No Root Index Page), because I realized that it was failing for some cases

https://github.com/dotCMS/core/pull/25254/files#diff-fd1b15e21bad3b62669bd99251d1ad511d30dc97f0c81959b4a52ed521ad8dceR30

* Create a lot of test

https://github.com/dotCMS/core/pull/25254/files#diff-fd1b15e21bad3b62669bd99251d1ad511d30dc97f0c81959b4a52ed521ad8dceR30

https://github.com/dotCMS/core/pull/25254/files#diff-291db88ba5f743045e03d89edf43c21ae3b52e23f0de62f2302907fd086f7aaaR20

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIntegrationTest.java [hunk 1]
```diff
@@ -133,7 +138,7 @@ public static void prepare() throws Exception {
      * - Second one in RUNNING state, it was already started and it was not stopped yet
      * - Finally one in STOP State, it was started, but it was stopped already
      */
-    @Test
+    //@Test
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIntegrationTest.java:141]: why?

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/IndexRegexUrlPatterStrategyIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+        assertFalse(("https://localhost:8080/?variantName=dotexperiment-fcaef4575a-variant-1&redirect=true").matches(regexPattern));
+        assertFalse(("https://localhost/?variantName=dotexperiment-fcaef4575a-variant-1&redirect=true").matches(regexPattern));
+    }
+
+
+
+    /**
+     * Method to test: {@link IndexRegexUrlPatterStrategy#isMatch(HTMLPageAsset)} and
+     * {@link IndexRegexUrlPatterStrategy#getRegexPattern(HTMLPageAsset)}
+     * When: The page an index page but not the root Index Page and the Domain is not localhost
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/experiments/business/IndexRegexUrlPatterStrategyIntegrationTest.java:222]: The page is an index page*

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/experiments/business/RegexUrlPatterStrategy.java [hunk 3]
```diff
@@ -21,7 +21,11 @@
  * - If the page is a Detail Page for a Content Type then we need to take account that the URL can be different.
  */
 abstract class RegexUrlPatterStrategy {
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/experiments/business/RegexUrlPatterStrategy.java:23]: @freddyucv This is a candidate to be an interface.

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIntegrationTest.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIntegrationTest.java]: uncomment https://github.com/dotCMS/core/pull/25254/commits/f907cd967b1dc06ef0cd95a90285f0d1753ef943#diff-34e7ad2b2aec5cebbf7ae6bb1697bff210c96c602184a9237bfda1a4c53e3c13R141-R2990

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIntegrationTest.java [hunk 4]
```diff
@@ -276,7 +281,7 @@ public void testStartExperiment_shouldPublishContent()
      * When: an {@link Experiment} is started
      * Should: publish all the contents in the variants created for the experiment.
      */
-    @Test(expected = IllegalArgumentException.class)
+    //@Test(expected = IllegalArgumentException.class)
```
[the remark this answers is not in the document]
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIntegrationTest.java]: uncomment https://github.com/dotCMS/core/commit/f907cd967b1dc06ef0cd95a90285f0d1753ef943#diff-34e7ad2b2aec5cebbf7ae6bb1697bff210c96c602184a9237bfda1a4c53e3c13R141-R2990

File: dotCMS/src/main/java/com/dotcms/experiments/business/RegexUrlPatterStrategy.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/experiments/business/RegexUrlPatterStrategy.java]: done https://github.com/dotCMS/core/pull/25254/commits/f907cd967b1dc06ef0cd95a90285f0d1753ef943#diff-0e580f38c58875e09d3247c7fd86090d1237652e2c5e257a53a8a27a799fbb6aR23

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/IndexRegexUrlPatterStrategyIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+        assertTrue(("http://localhost" + folder.getPath() + "/?variantName=dotexperiment-fcaef4575a-variant-1&redirect=true").matches(regexPattern));
+
+        assertFalse(("http://localhost:8080/?variantName=dotexperiment-fcaef4575a-variant-1&redirect=true").matches(regexPattern));
+        assertFalse(("http://localhost/?variantName=dotexperiment-fcaef4575a-variant-1&redirect=true").matches(regexPattern));
+    }
+
+    /**
+     * Method to test: {@link IndexRegexUrlPatterStrategy#isMatch(HTMLPageAsset)} and
+     * {@link IndexRegexUrlPatterStrategy#getRegexPattern(HTMLPageAsset)}
+     * When: The page is the an index page and the protocol is https
```
[the remark this answers is not in the document]
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/experiments/business/IndexRegexUrlPatterStrategyIntegrationTest.java]: done https://github.com/dotCMS/core/pull/25254/commits/3da387f10dc7fec73ab865b18be94e0739670abb

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/IndexRegexUrlPatterStrategyIntegrationTest.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/experiments/business/IndexRegexUrlPatterStrategyIntegrationTest.java]: done https://github.com/dotCMS/core/pull/25254/commits/3da387f10dc7fec73ab865b18be94e0739670abb

**Review by victoralfaro-dotcms — APPROVED** [review]

**Pull request merged**